### PR TITLE
Migrating extendedset from Metamarkets.

### PR DESCRIPTION
**Pull request by akashdw**
Migrating extendedset from metamarkets. This migration will be helpful to deal problems around having separate PRs and release cycle of Druid and extendedset . This PR is continuation of https://github.com/druid-io/druid/pull/3647.

Note:
1.  External extensions might have to change.
2. Differing license headers because `extendedset` is actually a fork of someone else's repository and `licensed to Metamarkets under a CLA` is not true.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -44,3 +44,11 @@ This product contains a modified version of Metamarkets bytebuffer-collections l
     * https://github.com/metamx/bytebuffer-collections
   * COMMIT TAG:
     * https://github.com/metamx/bytebuffer-collections/commit/3d1e7c8
+
+This product contains a modified version of Metamarkets extendedset library
```
Comment (gianm) [inline NOTICE]: This should probably mention the original codebase too

**Comment by fjy** [comment]
This needs to attribute the original codebase

**Comment by akashdw** [comment]
@fjy @gianm Found this url `http://concise.svn.sourceforge.net/viewvc/concise?view=rev&revision=1` from concise paper. Unfortunately this link is broken and returning 404. 

Code repo `https://sourceforge.net/projects/concise/` is also empty.
Please let me know if I'm missing anything or looking in the wrong place.

Also, what do you think adding below in NOTICE (assuming original code is missing from sourceforge)

> This product contains a modified version of Alessandro Colantonio's CONCISE
> (COmpressed 'N' Composable Integer SEt) library, extending the functionality of
> ConciseSet to use IntBuffers.
>  * (c) 2010 Alessandro Colantonio
>  * <mailto:colanton@mat.uniroma3.it>
>  * <http://ricerca.mat.uniroma3.it/users/colanton>

File: NOTICE (hunk 1 above)
Comment (akashdw) [inline NOTICE]: Updated the NOTICE.

**Comment by akashdw** [comment]
@fjy @gianm Updated the `notice` to attribute more details of original codebase and license.

**Comment by fjy** [comment]
👍

**Comment by gianm** [comment]
NB https://github.com/metamx/extendedset/pull/10 was merged in a branch upstream after this PR was filed.

**Comment by leventov** [comment]
@gianm no need to rebase, I'll PR into Druid after this PR (#3694) is merged.

**Comment by drcrallen** [comment]
@akashdw sorry for the delay, I think we're good here. If you can fix merge conflicts I can merge

**Comment by akashdw** [comment]
@drcrallen I added a suppress header check on extendedset files.

**Pull request merged**